grpcbp/prometheus_test.go

```diff
@@ -0,0 +1,57 @@
+package grpcbp
+
+import (
+	"testing"
+)
+
+func TestServiceAndMethodSlug(t *testing.T) {
+	testCases := []struct {
+		name        string
+		fullMethod  string
+		wantService string
+		wantMethod  string
+	}{
+		{
+			name:        "success",
+			fullMethod:  "/package.service/method",
+			wantService: "package.service",
+			wantMethod:  "method",
+		},
+		{
+			name:        "success 2",
+			fullMethod:  "package.service/method",
+			wantService: "package.service",
+			wantMethod:  "method",
+		},
+		{
+			name:        "extra /",
+			fullMethod:  "/foo/bar/baz",
+			wantService: "foo",
+			wantMethod:  "bar/baz",
+		},
+		{
+			name:        "no /",
+			fullMethod:  "package.service.method",
+			wantService: "",
```

Comment:
since this is for monitoring, you want to fail closed.  In this case, that probably means that when you don't recognize the format, you stick it all in the "method" label, since that's the most specific label and most likely matches the expected cardinality.